### DISCUSSION TITLE
liburing testing suite integration inside kirk (All-in-one Testing Framework)

### DISCUSSION THREAD

**Discussion by acerv** [axboe/liburing#865] [Show and tell]
Hello,

my name is Andrea Cervesato, QA Kernel Engineer at [SUSE](https://www.suse.com/), and I'm the main developer of [kirk](https://github.com/acerv/kirk), an All-in-one runner for testing frameworks, created with the idea of testing Linux kernel using multiple frameworks (also via SSH, Qemu, etc). The project started including [LTP](https://github.com/linux-test-project/ltp) and moved forward including other testing frameworks.

In our QA team, we found liburing testing suite a nice way to test Linux io_uring support and for this reason I decided to include it in the kirk base code. You can find the source code [here](https://github.com/acerv/kirk/blob/master/libkirk/liburing.py) and maybe find it interesting for your project.

Thanks,
Andrea

**Comment by axboe** [comment]
That sounds great! For me personally, I run the tests manually, and I keep wanting to setup something that'll test branches from my kernel tree whenever changes happen. Maybe I should look into kirk for that?

**Reply by acerv** [reply]
kirk works on running Linux kernel, but it's possible to configure its internal Qemu communication in order to boot busybox images. This can be done due to the Qemu `-kernel` option. I was planning to write documentation about it.
If you are interested I can share it in the next future.
